packages/block-editor/src/components/colors-gradients/tools-panel-color-dropdown.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	ColorIndicator,
+	Dropdown,
+	FlexItem,
+	__experimentalHStack as HStack,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import ColorGradientControl from './control';
+import useMultipleOriginColorsAndGradients from './use-multiple-origin-colors-and-gradients';
+
+export default function ToolsPanelColorDropdown( {
```

Comment:
Why did we introduce a new component instead of updating the existing `ColorGradientSettingsDropdown`? Now basically we have two concurrent components for developers to use and they have duplicated styles and very similar behavior.

cc @jorgefilipecosta as we discussed this a lot when we first updated `ColorGradientSettingsDropdown` to use the global styles like UI

Reply:
This PR was already open and underway before the color controls got updated. It turned out they were updated without taking into account the future use case where they would be within the `ToolsPanel`. At this point we have two different needs for a color control. One that can be used directly in a limited `ItemGroup` and another that can exist within a `ToolsPanel`.

The `ColorGradientSettingsDropdown` renders a group of color controls within an `ItemGroup`. This can't really be done when the possible color controls are being rendered via the block support SlotFills and to maintain ordering of `ToolsPanel` controls placeholders must be rendered which interferes with `ItemGroup` styling and layout. There are also no restrictions on what can be rendered via those slots meaning it could contain additional help text, contrast checkers and so on.

The earlier comments and conversation on this PR also provide some background as to the issues faced as well as alternative approaches explored such as attempting to make the color block support slot render as an `ItemGroup` etc.

Reply:
Can't we just replace the ItemGroup version with this one which if I'm understanding properly is the same visually, but a bit different in terms of component tree in order to allow extensibility?

Reply:
I'm not sure we can, or at least, not directly.

The `ItemGroup` based version allows for a full set of color controls to be grouped semantically. The separate version added here was for when that wasn't possible i.e. within the context of a `ToolsPanel`. It also allowed for the controls to be rendered individually rather than as part of a group e.g. injecting an individual color control into the color block supports panel.

The original `ItemGroup` version is still in use in the Cover block, for its opacity controls via `PanelColorGradientSettings`, as well as possibly by third party plugins etc. Also, `PanelColorGradientSettings` has a native implementation that I don't think makes sense for the dropdown component introduced in this PR given the `ToolsPanel` is only used on the web.

Apologies if I'm misunderstanding the question.

Reply:
> Maybe I'm missing something here as well :)

I think it's more likely confusion on my part.  Thanks for your patience. 🙏 

> From a user's perspective, both seem to have the same output. Semantics seem internal implementation details.

I can agree that visually, the output from a single `ColorGradientSettingsDropdown` passed an array of settings objects will look much the same as a series of individual `ToolsPanelColorDropdown` components rendered in the Color block support `ToolsPanel`. 

That was intended so the color controls within the block support `ToolsPanel` still matched the design / current appearance.

Another point of confusion could be the name of `ColorGradientSettingsDropdown` is misleading as well given it actually renders more than a single dropdown. A better name might be `ColorGradientSettingsItemGroup`.

> From a user perspective, it seems to me like PanelColorGradientSettings is a special case (no extensibility version) of ToolsPanelColorDropdown.

Unfortunately, I'm not following this comment. 

To me they do different things. The `PanelColorGradientSettings` wraps a `ColorGradientSettingsDropdown` in a `PanelBody` so we end up with a fixed standalone panel like:

```
<PanelBody>
  <ItemGroup>
    <Item>
      <Dropdown>
    </Item>
    <Item>
      <Dropdown>
    </Item>
    ...
  </ItemGroup>
</PanelBody>
```

Whereas the `ToolsPanelColorDropdown` only gives us a `ToolsPanelItem` that can be rendered as a fill where desired:

```
<ToolsPanelItem>
    <Dropdown>
</ToolsPanelItem>
```

This allows that to be injected into our color block support panel resulting in:

```
<ToolsPanel>
  <ToolsPanelItem>
      <Dropdown>
  </ToolsPanelItem>
  ...
</ToolsPanel>
```

> Can we make the ToolsPanelColorDropdown API (props) to match PanelColorGradientSettings in which case, there's no impact on third-party or core usage (both functionality wise since it's just an addition of extensibility, and in terms of code since the APIs match).

I could see more of a possibility of aligning the `ColorGradientSettingsDropdown` and the `ToolsPanelColorDropdown`.  This was what I thought you were originally suggesting.

Merging these two components would be achievable but in my view, it overcomplicates the resulting component and its use by consumers. I appreciate that once introduced we need to support this new component so I'm open to refactoring this.

All that said, I acknowledge there is still the problem of the shared styles.

Reply:
Thanks for clarifying more it makes a lot of sense.

> I could see more of a possibility of aligning the ColorGradientSettingsDropdown and the ToolsPanelColorDropdown. This was what I thought you were originally suggesting.

Yeah for me, I was discussing the "whole thing" as well, not a single item so I probably got confused in the naming as well. 

--
Let me try to clarify how I end up here, and I think it will help you understand my concerns and allow us (probably you since you know more than me about these components) find a path forward.

I wanted to ask a button (reset icon) on the right of each item in the color panel. The button would serve as a way to "reset/clear the value of the given item without the need to reach out to the three dots menu. While doing so, I noticed that these color buttons are duplicated in two different components and wasn't sure which one I should be using/updating (probably both), so I thought that there might not be a reason for these "color panel items" to be duplicated.

Reply:
> I wanted to ask a button (reset icon) on the right of each item in the color panel. The button would serve as a way to "reset/clear the value of the given item without the need to reach out to the three dots menu. 

As you note, the Color panel in the inspector controls sidebar is a `ToolsPanel` with an ellipsis menu. If it is only in this panel, you wish to add the reset button, it could be added to the `ToolsPanelColorDropdown`. I can also see why we'd want to be consistent and add it to the controls when they aren't within a `ToolsPanel` as well.

One thing I'd note here is that a goal for the `ToolsPanel` was to declutter the sidebar removing the repeated reset/clear buttons everywhere. @jasmussen might be better positioned to expand on this. My knowledge on current design goals there could also be out of date.

> so I thought that there might not be a reason for these "color panel items" to be duplicated.

Understandable. Hopefully, the extra information in my earlier comments illustrates that while there's overlap, they serve different needs. 

I can draft a PR that explores merging these components. I'm expecting the result to be less than desirable though.

Reply:
Thank you for the ping. Yes indeed, that's still a task that needs careful thought. One first step I've explored is a reset button on the itemgroups, like Riad suggests:

<img width="1536" alt="Clearing" src="https://user-images.githubusercontent.com/1204802/161701087-2ee44358-86b9-4609-94c2-bbed29310887.png">

This would share some DNA with ideas outlined in #39427.

A quicker/simpler fix might be to re-introduce the reset button inside the color popover. @aaronrobertshaw I recall you had one in the border popover, right? As we explore options to clarify the whole flow, a simple clear button like that inside all color popovers might be fine.

I'd make it a tertiary not-small style button though. What do you think?

Reply:
> I'd make it a tertiary not-small style button though. What do you think?

Thanks for the designs @jasmussen, that makes sense now. I'm not opposed to the unobtrusive tertiary button described. Anything that helps sort out the UX around resetting values is fine by me 👍 

> A quicker/simpler fix might be to re-introduce the reset button inside the color popover. @aaronrobertshaw I recall you had one in the border popover, right?

There is a "Reset to default" button in the new `BorderControl`'s color and style picker that will clear the border color and style values.

Reply:
I've created a PR (https://github.com/WordPress/gutenberg/pull/40084) to merge the two color dropdown related components. I'll resolve this discussion so it can be continued there with tangible code changes to review.

Reply:
Thanks Aaron! 

And just to be clearery about the clear button, I was thinking that instead of this:
<img width="282" alt="Screenshot 2022-04-06 at 09 58 00" src="https://user-images.githubusercontent.com/1204802/161925321-84ea6093-ff3e-4938-b6cf-0070a1266935.png">

we do this:

<img width="282" alt="Screenshot 2022-04-06 at 09 57 43" src="https://user-images.githubusercontent.com/1204802/161925339-5de029b6-0b08-461a-a2a5-b41809186572.png">